streams/src/test/java/org/apache/kafka/streams/integration/InternalTopicIntegrationTest.java

```diff
@@ -135,6 +141,37 @@ private Admin createAdminClient() {
         return Admin.create(adminClientConfig);
     }
 
+    /*
+     * This test just ensures that that the assignor does not get stuck during partition number resolution
+     *  for internal repartition topics. See KAFKA-10689
+     */
```

Comment:
Do we need this comment?

Reply:
I'd prefer to leave it in because otherwise the test looks totally out of place and doesn't seem to test anything. Historical context can be useful 🙂

Reply:
To quote a [very salty developer](https://pastebin.com/PTLeWhc2) in the comments of the leaked Windows Media Player source code:

```
/* The reason for requiring the following test is now lost
         * in the mists of time.  Now this app is 32-bit, these
         * bogus timer callbacks (if they really do still occur)
         * could be 16-bit, so we need to add yet more ugliness
         * in the form of assembler to an app which is already
         * hardly a paragon of pulchritude.
         *
         * A plea:
         *
         * If you add some obscure code such as below, to this or
         * any other app, even if it has only the teeniest chance
         * of being less blindingly obvious to someone else than
         * it is to you at the time of writing, please please please
         * add a f***ing comment.
         *
         * Respectfully,
         * A Developer
         */
```
😂